lib/cucumber/formatter/junit.rb

```diff
@@ -25,155 +21,134 @@ def initialize(runtime, io, options)
         @options = options
       end
 
-      def before_feature(feature)
-        @current_feature = feature
-        @failures = @errors = @tests = @skipped = 0
-        @builder = Builder::XmlMarkup.new( :indent => 2 )
-        @time = 0
+      def before_test_case(test_case)
+        if not same_feature_as_previous_test_case?(test_case.feature)
```

Comment:
Yes, you're right, in this case there is no `else`, and we are free to improve one step more.

(By preferring `unless` over `if !` and by not using `not` you now fixed two things.)